scripts/zkai_native_d128_verifier_execution_compression_budget_gate.py

```
@@ -0,0 +1,679 @@
+        raise AssertionError(f"unhandled mutation {name}")
+    mutated["payload_commitment"] = payload_commitment(mutated)
+    return mutated
+
+
+def collect_mutation_rejections(cases: tuple[dict[str, Any], ...]) -> list[dict[str, Any]]:
+    rejected = []
+    for case in cases:
+        try:
+            validate_payload(case["payload"], allow_missing_mutation_summary=True)
+        except Exception as err:
+            rejected.append({"name": case["name"], "rejected": True, "error": str(err)})
```

Ruff (0.15.12)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)